homeassistant/helpers/update_coordinator.py

```diff
@@ -84,7 +86,12 @@ def __init__(
         self._update_interval_seconds: float | None = None
         self.update_interval = update_interval
         self._shutdown_requested = False
-        self.config_entry = config_entries.current_entry.get()
+        if config_entry is UNDEFINED:
+            self.config_entry = config_entries.current_entry.get()
+            # This should be deprecated once all core integrations are updated
+            # to pass in the config entry explicitly.
+        else:
+            self.config_entry = config_entry
```

Comment:
I think we should raise `ValueError` if `self.config_entry` is None in `async_config_entry_first_refresh`.

Reply:
> I think we should raise `ValueError` if `self.config_entry` is None in `async_config_entry_first_refresh`.

Done - with tests